Source/WebCore/css/CSSGroupingRule.h

```diff
@@ -46,7 +46,14 @@ class CSSGroupingRule : public CSSRule {
     const StyleRuleGroup& groupRule() const { return m_groupRule; }
     StyleRuleGroup& groupRule() { return m_groupRule; }
     void reattach(StyleRuleBase&) override;
-    void appendCSSTextForItems(StringBuilder&) const;
+    String cssText() const override;
+
+    // https://drafts.csswg.org/cssom/#serialize-a-css-rule
+    struct CSSText {
+        StringBuilder decls;
+        StringBuilder rules;
+    };
+    CSSText cssTextForItems() const;
```

Comment:
Returning StringBuilders is odd. They are only used to extract strings so they could just be Strings.

It would be more efficient to provide StringBuilders as arguments that cssTextForItems would write to.

"Items" is vague.

Reply:
`items` is also CSSOM spec vocabulary.

I will change to pass StringBuilder as arguments.